### PR TITLE
feat: make `add_lags` dataframe-agnostic

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Before working on a large PR, please check with @FBruzzesi or @koaning to confirm that they agree with the direction of the PR. This discussion should take place in a [Github issue](https://github.com/koaning/scikit-lego/issues/new/choose) before working on the PR, unless it's a minor change like spelling in the docs. 

# Description

[add_lags](https://koaning.github.io/scikit-lego/api/pandas-utils/?h=add_lags) is now dataframe-agnostic - this one was surprisingly easy

Fixes #(issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the style guidelines (ruff)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (also to the readme.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added tests to check whether the new feature adheres to the sklearn convention
- [ ] New and existing unit tests pass locally with my changes

If you feel your PR is ready for a review, ping @FBruzzesi or @koaning.

### CODE REVIEW

File: sklego/pandas_utils.py [hunk 1]
```diff
@@ -332,23 +347,19 @@ def _add_lagged_pandas_columns(df, cols, lags, drop_na):
...
-    # Indexes are not supported as pandas column names may be
-    # integers themselves, introducing unexpected behaviour
-    if not all([col in df.columns.values for col in cols]):
+    if not all([col in df.columns for col in cols]):
         raise KeyError("The column does not exist")
 
-    combos = (df[col].shift(-lag).rename(col + str(lag)) for col in cols for lag in lags)
+    answer = df.with_columns([nw.col(col).shift(-lag).alias(col + str(lag)) for col in cols for lag in lags])
 
-    answer = pd.concat([df, *combos], axis=1)
```
Comment (MarcoGorelli) [inline sklego/pandas_utils.py:348]: this diff is surprisingly pleasing

File: tests/test_pandas_utils/test_pandas_utils.py [hunk 2]
```diff
@@ -62,8 +70,9 @@ def test_add_lagged_numpy_columns(test_X):
         _add_lagged_numpy_columns(test_X, ["test"], 1, True)
 
 
-def test_log_step(capsys, test_df):
+def test_log_step(capsys, data):
     """Base test of log_step without any arguments to the logger"""
+    test_df = pd.DataFrame(data)
```
Comment (MarcoGorelli) [inline tests/test_pandas_utils/test_pandas_utils.py:75]: I haven't worked on `log_step` in this PR, so for now I'm just keeping the `test_log_step` tests to pandas

**Comment by MarcoGorelli** [comment]
> ```
> FAILED tests/test_estimators/test_demographic_parity.py::test_standard_checks[check_fit2d_predict1d] - TypeError: Clarabel: unrecognized solver setting 'max_iters'.
> ```

🤔 not sure what's going on with tests in CI

File: sklego/pandas_utils.py [hunk 3]
```diff
@@ -332,23 +347,19 @@ def _add_lagged_pandas_columns(df, cols, lags, drop_na):
...
     cols = as_list(cols)
 
-    # Indexes are not supported as pandas column names may be
-    # integers themselves, introducing unexpected behaviour
-    if not all([col in df.columns.values for col in cols]):
+    if not all([col in df.columns for col in cols]):
         raise KeyError("The column does not exist")
 
-    combos = (df[col].shift(-lag).rename(col + str(lag)) for col in cols for lag in lags)
+    answer = df.with_columns(nw.col(col).shift(-lag).alias(col + str(lag)) for col in cols for lag in lags)
```
Comment (FBruzzesi) [inline sklego/pandas_utils.py:359]: Any plan for the `.name` namespace? Asking for a friend 😂
I can take a look myself during the weekend 😇

Comment (MarcoGorelli) [reply sklego/pandas_utils.py:359]: definitely!

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -28,6 +28,6 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
-      run: python -m pip install -e ".[test-all]"
+      run: python -m pip install "cvxpy<1.5.0" -e ".[test-all]"
```
Comment (FBruzzesi) [inline .github/workflows/test.yml:31]: If you merge main there should be no need for this pinning now

**Review by FBruzzesi — APPROVED** [review]
Thanks Marco 👋🏼 ! This is moving fast!

FlashRunGIF

**Pull request merged**